### ISSUE TITLE
Terraform should create if it doesnt exist but instead there are “not found” errors being returned.

### ISSUE THREAD

**Issue by mmalave1** [newrelic/terraform-provider-newrelic#860]
I deleted it to see if recreating one would fix the issue posted here: [Alert fired after alerts refreshed in terraform](https://discuss.newrelic.com/t/alert-fired-after-alerts-refreshed-in-terraform/113224)

Terraform should create if it doesnt exist but instead there are “not found” errors being returned.

`Error: A downstream error occurred., [{"extensions":{"code":"BAD_USER_INPUT"},"message":"Not Found"}]
 ERROR: Job failed: exit code 1
`

**Comment by sanderblue** [comment]
Hi @mmalave1, we need to see your configuration (sensitive info redacted) and a little more context to better understand whats going on. 

As a side note, if you're reporting a potential bug, etc please follow our issue template guidelines when submitting issues. This will expedite resolving the issue. 🙂

**Comment by mmalave1** [comment]
It seems this failure doesnt occur when an infra alert is deleted but it does occur when an nrql alert is. 

Here are the request details for the request while the API key in the header is the one that is expected for that account:

```
{
	"data": {
		"actor": {
			"account": {
				"alerts": {
					"nrqlCondition": null
				}
			}
		}
	},
	"errors": [{
		"downstreamResponse": [{
			"extensions": {
				"code": "BAD_USER_INPUT"
			},
			"locations": [{
				"column": 5,
				"line": 4
			}],
			"message": "Not Found",
			"path": ["account", "nrqlCondition"]
		}],
		"id": 106351049,
		"locations": [{
			"column": 7,
			"line": 6
		}],
		"message": "A downstream error occurred.",
		"path": ["actor", "account", "alerts", "nrqlCondition"]
	}]
}
```

**Comment by sanderblue** [comment]
So I went on a hunch and was able to reproduce this locally with a minimal configuration that includes an alert policy and a NRQL condition, then deleted the condition manually from the UI and tried to rerun `terraform apply`. This definitely seems odd, so we'll look into it. 

For reference, here is the example config that reproduces the error for me 
```hcl
provider "newrelic" {}

resource "newrelic_nrql_alert_condition" "test" {
  policy_id            = var.test_policy_id
  name                 = "tf-test-sblue-recreate-error"
  runbook_url          = ""
  violation_time_limit = "twenty_four_hours"
  value_function       = "single_value"

  critical {
    threshold_duration    = 240
    operator              = "above"
    threshold             = 90
    threshold_occurrences = "all"
  }

  nrql {
    query             = "SELECT count(*) FROM TransactionError WHERE appName like '%Dummy App%'"
    evaluation_offset = 3
  }
}
```
> **Note:** Providing a sample configuration when filing a bug will help expedite debugging the issue 🙂 

I received the same error
```
Error: A downstream error occurred., [{"extensions":{"code":"BAD_USER_INPUT"},"message":"Not Found"}]
```

**Comment by sanderblue** [comment]
This has been fixed in **v2.7.2** 🙂
